src/actor/mod.rs

```diff
@@ -34,3 +51,25 @@ async fn run_actor<T: Send + Sync>(mut actor: Box<dyn Actor<T> + Send + Sync>) {
     actor.handle_msg(msg).await
   }
 }
+
+async fn run_actor_with_shutdown<T: Send + Sync>(
+  mut actor: Box<dyn Actor<T> + Send + Sync>,
+  shutdown_token: CancellationToken,
+) {
+  loop {
```

Comment:
Is this an efficiency issue? This loop might run hot, where-as `run` yielded on each iteration.